data/ai/lua/generic_recruit_engine.lua

```diff
@@ -650,11 +661,29 @@ return {
             local best_scores = {offense = 0, defense = 0, move = 0}
             local best_hex = recruit_data.recruit.best_hex
             local target_hex = recruit_data.recruit.target_hex
-            local distance_to_enemy, enemy_location
-            if target_hex[1] then
-                distance_to_enemy, enemy_location = AH.get_closest_enemy(target_hex, wesnoth.current.side, { viewing_side = 0 })
-            else
-                distance_to_enemy, enemy_location = AH.get_closest_enemy(best_hex, wesnoth.current.side, { viewing_side = 0 })
+
+            local reference_hex = target_hex[1] and target_hex or best_hex
+            local distance_to_enemy, enemy_location = AH.get_closest_enemy(reference_hex, wesnoth.current.side, { viewing_side = 0 })
+            -- If no enemy is on the map, then we first use closest enemy start hex,
+            -- and if the does not exist either, a location mirrored w.r.t the center of the map
+            if not enemy_location then
+                local enemy_sides = wesnoth.get_sides({ { "enemy_of", {side = wesnoth.current.side} } })
+                local min_dist
```

Comment:
Initialize to a large value to make later comparison simpler?

Reply:
Funny thing about that, I've had somebody tell me that what you are suggesting is bad style ...  I personally don't really care one way or the other, so I'll initialize it to the usual 9e99.

Reply:
How odd. The style guides at work all say to initialize variables at the point of declaration if possible, though they allow for non-explicit initialization if the default value is present and the one you want.

Since they're publicly available, I present the Javascript one as an example (as a language that is close to Lua): https://google.github.io/styleguide/jsguide.html#features-declared-when-needed

Reply:
Thanks, I'll poke around that a bit in general.  IIRC the complaint was not so much about initializing the variable per se, but about using a large number as a proxy for infinity, with what I had done suggested as a better solution.

Reply:
Brief testing in the Lua REPL reveals that `9e999` is too large to represent and becomes infinity, so perhaps you could use that instead? Though I think it'd still be better to actually have a constant named `infinity` that you can use in these situations.

Reply:
In that case, use math.huge

Reply:
Yup, `math.huge` works in the Lua REPL. We do expose the Lua math library and don't fiddle with it, so that should work in Wesnoth too.